app/lib/.server/llm/create-summary.ts

```diff
@@ -0,0 +1,138 @@
+  let currentModel = DEFAULT_MODEL;
+  let currentProvider = DEFAULT_PROVIDER.name;
+  const processedMessages = messages.map((message) => {
+    if (message.role === 'user') {
+      const { model, provider, content } = extractPropertiesFromMessage(message);
+      currentModel = model;
+      currentProvider = provider;
+
+      return { ...message, content };
+    } else if (message.role == 'assistant') {
+      let content = message.content;
+
+      if (contextOptimization) {
+        content = simplifyBoltActions(content);
+      }
+
+      return { ...message, content };
+    }
+
+    return message;
+  });
+
+  const provider = PROVIDER_LIST.find((p) => p.name === currentProvider) || DEFAULT_PROVIDER;
+  const staticModels = LLMManager.getInstance().getStaticModelListFromProvider(provider);
+  let modelDetails = staticModels.find((m) => m.name === currentModel);
+
+  if (!modelDetails) {
+    const modelsList = [
+      ...(provider.staticModels || []),
+      ...(await LLMManager.getInstance().getModelListFromProvider(provider, {
+        apiKeys,
+        providerSettings,
+        serverEnv: serverEnv as any,
+      })),
+    ];
+
+    if (!modelsList.length) {
+      throw new Error(`No models found for provider ${provider.name}`);
+    }
+
+    modelDetails = modelsList.find((m) => m.name === currentModel);
+
+    if (!modelDetails) {
+      // Fallback to first model
+      logger.warn(
+        `MODEL [${currentModel}] not found in provider [${provider.name}]. Falling back to first model. ${modelsList[0].name}`,
+      );
+      modelDetails = modelsList[0];
+    }
+  }
+
+  let slicedMessages = processedMessages;
+  const { summary } = extractCurrentContext(processedMessages);
+  let summaryText: string | undefined = undefined;
+  let chatId: string | undefined = undefined;
+
+  if (summary && summary.type === 'chatSummary') {
+    chatId = summary.chatId;
+    summaryText = `Below is the Chat Summary till now, this is chat summary before the conversation provided by the user 
+you should also use this as historical message while providing the response to the user.        
+${summary.summary}`;
+
+    if (chatId) {
+      let index = 0;
+
+      for (let i = 0; i < processedMessages.length; i++) {
+        if (processedMessages[i].id === chatId) {
+          index = i;
+          break;
+        }
+      }
+      slicedMessages = processedMessages.slice(index + 1);
+    }
+  }
+
+  const extractTextContent = (message: Message) =>
+    Array.isArray(message.content)
+      ? (message.content.find((item) => item.type === 'text')?.text as string) || ''
+      : message.content;
+
+  // select files from the list of code file from the project that might be useful for the current request from the user
+  const resp = await generateText({
+    system: `
+        You are a software engineer. You are working on a project. tou need to summarize the work till now and provide a summary of the chat till now.
```

Comment:
I may not have fully understood, but shouldn't we add the initial message always? Or maybe instruct the model to always keep this as intent?

Reply:
no every-time you call the API you have to pass all the messages like this.
it starts with an empty slate on each call

Reply:
Sure! But the first message in a bolt-conversation is like the users intent. It usually features restrictions about tech to use etc. thus, I think it is special and we shouldn't redact information from it.

Reply:
we are not redacting that.. its present in the existing summary.
so the summary generation is incremental.
instead of summarizing the entire chat on each request, i made it like summarize the chat using the previous summary and the new chats added after that.

so the all the initial chat and every key details show be present in the previous summary